lib/java_buildpack/framework/contrast_security_agent.rb

```diff
@@ -0,0 +1,114 @@
+# Encoding: utf-8
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2017 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'fileutils'
+require 'open-uri'
+require 'rexml/document'
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/framework'
+require 'base64'
+
+module JavaBuildpack
+  module Framework
+
+    # Encapsulates the functionality for running the Contrast Security Agent support.
+    class ContrastSecurityAgent < JavaBuildpack::Component::VersionedDependencyComponent
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        load_credentials
+        FileUtils.mkdir_p(@droplet.sandbox)
+        @dir = @droplet.sandbox
+        download_jar(boot_class_name)
+        build_contrast_configuration
+        @droplet.copy_resources
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#release)
+      def release
+        app_name = @application.details['application_name'] || 'ROOT'
+        java_opts = @droplet.java_opts
+        java_opts.add_system_property('contrast.dir', '/tmp')
+        java_opts.add_system_property('contrast.override.appname', app_name)
+        path = java_opts.qualify_path(@droplet.sandbox)
+        java_opts.add_preformatted_options("-javaagent:#{path}/#{boot_class_name}=#{path}/contrast.config")
+      end
+
+      protected
+
+      # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
+      def supports?
+        @application.services.one_service?(CONTRAST_FILTER, [TEAMSERVER_URL, USERNAME, API_KEY, SERVICE_KEY])
+      end
+
+      private
+
+      CONTRAST_FILTER = /contrast[-]?security/
```

Comment:
Since this the first time we've ever integrated, you should pick a single, canonical, representation that you'd like to search for and that should become the truth.